compiler/cpp/src/thrift/generate/t_js_generator.cc

```diff
@@ -1580,10 +1580,10 @@ void t_js_generator::generate_service_client(t_service* tservice) {
       } else {
         f_service_ << indent() << "if (callback) {" << endl;
         f_service_ << indent() << "  var self = this;" << endl;
-        f_service_ << indent() << "  this.output.getTransport().flush(true, function() {" << endl;
+        f_service_ << indent() << "  this.output.getTransport().flush(true, function(xhrError) {" << endl;
         f_service_ << indent() << "    var result = null;" << endl;
         f_service_ << indent() << "    try {" << endl;
-        f_service_ << indent() << "      result = self.recv_" << funname << "();" << endl;
+        f_service_ << indent() << "      result = self.recv_" << funname << "() || xhrError;" << endl;
```

Comment:
A few questions/comments:

In general, why is this so different than the gen_jquery_?
in the context of an xhrError, what is the expected behavior of the recv_xyz() function? Should it ever return anything? Does it have side effects that we need? If not, should we do xhrError || self.recv_xyz()?

Tangent: This seems like the place where you would potentially wrap the result in a promise if you were so inclined...and it almost seems reasonable to do at some point?